### PR TITLE
✨ create_commit: more user-friendly errors on HTTP 400

### PR DESCRIPTION
**Pull request by SBrandeis**
Related to #962 

This PR converts HTTP 400 errors from `/api/preupload` and `/api/commit` APIs into `ValueError`s

Suggestions/feedback welcome

### CODE REVIEW

**Review by adrinjalali — COMMENTED** [review]
This looks nice, but it might cause some friction if users are catching those exceptions.

I'd suggest an exception class which inherits from `ValueError` and `HTTPError` so that existing user code would work.

The methods which can raise this exception can document it as `ValueError`, but if users are already catching `HTTPError`, their code wouldn't break after an upgrade.

WDYT?

**Comment by SBrandeis** [comment]
That's a very valid point @adrinjalali, thanks for the suggestion!

File: src/huggingface_hub/_commit_api.py [hunk 1]
```diff
@@ -379,7 +380,7 @@ def fetch_upload_modes(
         json=payload,
         headers=headers,
     )
-    resp.raise_for_status()
+    _raise_convert_bad_request(resp, endpoint_name="preupload")
```
Comment (adrinjalali) [inline src/huggingface_hub/_commit_api.py:383]: The docstring needs to be fixed, where it says:

```
    Raises:
        :class:`requests.HTTPError`:
            If the Hub API returned an error
```

File: src/huggingface_hub/hf_api.py [hunk 2]
```diff
@@ -1916,7 +1920,7 @@ def create_commit(
             json=commit_payload,
             params={"create_pr": 1} if create_pr else None,
         )
-        _raise_for_status(commit_resp)
+        _raise_convert_bad_request(commit_resp, endpoint_name="commit")
```
Comment (adrinjalali) [inline src/huggingface_hub/hf_api.py:1923]: docstring missing `Raises` section / note.

**Review by adrinjalali — APPROVED** [review]
Otherwise LGTM.

**Comment by SBrandeis** [comment]
cc @Wauplin

File: src/huggingface_hub/hf_api.py [hunk 3]
```diff
@@ -1857,6 +1861,10 @@ def create_commit(
             `str` or `None`:
                 If `create_pr` is `True`, returns the URL to the newly created Pull Request
                 on the Hub. Otherwise returns `None`.
+
+        Raises:
+            :class:`ValueError`:
+                If the Hub API returns an HTTP 400 error (bad request)
```
Comment (Wauplin) [inline src/huggingface_hub/hf_api.py:1867]: Hey 👋 
I know it's not exactly related to this issue but I realized when reading the docs that the "Raises" part is never rendered by the autodoc (example: https://moon-ci-docs.huggingface.co/docs/huggingface_hub/pr_963/en/package_reference/hf_api#huggingface_hub.HfApi.create_commit). Is that a bug or a feature ? 🙄 

If a bug, I can create a separate issue on the repo to document it.

Comment (adrinjalali) [reply src/huggingface_hub/hf_api.py:1867]: cc @mishig25

Comment (Wauplin) [reply src/huggingface_hub/hf_api.py:1867]: Seems to have been discussed here as well so I guess we are fine to keep it as it is for now.

**Review by Wauplin — APPROVED** [review]

**Pull request merged**